libevmasm/Assembly.cpp

```diff
@@ -696,12 +698,10 @@ LinkerObject const& Assembly::assemble() const
 		}
 	}
 
-	assertThrow(
-		immutableReferencesBySub.empty(),
-		AssemblyException,
-		"Some immutables were read from but never assigned."
-	);
-
+	if (!immutableReferencesBySub.empty())
+		throw
+			langutil::Error(1284_error, langutil::Error::Type::CodeGenerationError) <<
+			util::errinfo_comment("Some immutables were read from but never assigned, possibly because of optimization.");
```

Comment:
We can throw `Error` or introduce another exception (and then construct `Error` right before reporting).

Throwing `Error` looks simpler, but we need to check how it works in combination with already existing handlers, e.g.:

```
	/// This is only thrown in a very few locations.
	catch (Error const& _error)
	{
		errors.append(formatErrorWithException(
			_error,
			false,
			_error.typeName(),
			"general",
			"Uncaught error: "
		));
	}
```